### PR TITLE
Refactor HA tests for multi-machine jobs

### PR DESCRIPTION
**Pull request by sysrich**

### CODE REVIEW

File: tests/ha/sle11_cluster_init.pm [hunk 1]
```diff
@@ -3,12 +3,34 @@ use testapi;
 use autotest;
 
 sub run() {
-    send_key 'shift-ctrl-alt-g';
-    type_string "sleha-init -y -s /dev/disk/by-id/scsi-1LIO-ORG.FILEIO:348cfd84-58a1-426e-9797-e22f04cf207f\n";
-    assert_screen 'cluster-init',60;
+    if (check_var("WORKER_CLASS", "ha_master")) {
```
Comment (coolo) [inline tests/ha/sle11_cluster_init.pm]: this is very dangerous as WORKER_CLASS is a list. You're better off with get_var('WORKER_CLASS') =~ /ha_master/

File: tests/ha/sle11_cluster_init.pm [hunk 2]
```diff
@@ -3,12 +3,34 @@ use testapi;
...
+        type_string "sleha-init -y -s /dev/disk/by-id/scsi-1LIO-ORG.FILEIO:348cfd84-58a1-426e-9797-e22f04cf207f\n";
+        assert_screen 'cluster-init',60;
+        type_string "crm status\n";
+        assert_screen 'cluster-status';
+        send_key 'ctrl-l';
+        mutex_create('cluster-init');
```
Comment (aaannz) [inline tests/ha/sle11_cluster_init.pm]: Does it work for you as is? mutex_create is exported by lockapi.pm which I don't see being imported. I'm missing 'use lockapi'.

**Comment by sysrich** [comment]
Fixes pushed, please merge :)

**Comment by coolo** [comment]
you seriously show typical "I'm a developer, QA will test this" symtoms - that code is not even perl :)

**Comment by aaannz** [comment]
I'm working on missing os-autoinst dependencies and trying to add compile checker to tests. Will post PR shortly to fix this.

**Comment by aaannz** [comment]
see os-autoinst/os-autoinst#230

**Comment by aaannz** [comment]
This needs to be rebased now to get new travis.yml in the branch.

**Comment by sysrich** [comment]
Okay, I think I made a real pigs ear of the rebase..but..it works now?

**Comment by aaannz** [comment]
how about this one #364

**Pull request closed without merging**